### PR TITLE
Add an option to report deltas during a flow period instead of cumulative stats

### PR DESCRIPTION
**Pull request by kvalliyurnatt**
## Proposed commit message

Currently intermediate flows and final flows report cumulative stats for `network.bytes` and `network.packets`. Adding a config option `enable_delta_flow_reports` that will result in intermediate flows and final flows reporting `network.bytes` and `network.packets` for only that corresponding period instead of cumulative.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Use cases
 Having delta stats in intermediate flows helps calculating things like total bytes for a specific  time window easier without having to look at flows for other time periods to arrive at an accurate answer.

### CODE REVIEW

File: packetbeat/docs/packetbeat-options.asciidoc [hunk 1]
```diff
@@ -461,6 +461,12 @@ in time. Periodical reporting can be disabled by setting the value to -1. If
 disabled, flows are still reported once being timed out. The default value is
 10s.
 
+[float]
+==== `enable_delta_flow_reports`
+
+Configure network.bytes and network.packets to be a delta
+value instead of a cummlative sum for each flow period. The default value is false.
```
Comment (yspotts) [inline packetbeat/docs/packetbeat-options.asciidoc:468]: I think there is only one `m`

File: packetbeat/flows/worker_test.go [hunk 2]
```diff
@@ -116,7 +116,7 @@ func TestCreateEvent(t *testing.T) {
 
 	// Write the event to disk if -data is used.
 	if *dataFlag {
-		event.Fields.Put("@timestamp", common.Time(end)) //nolint:errcheck // Never fails.
```
Comment (yspotts) [inline packetbeat/flows/worker_test.go:119]: why was this removed?

**Review by yspotts — COMMENTED** [review]
overall, looks good to me.

File: packetbeat/flows/worker_test.go (hunk 2 above)
Comment (kvalliyurnatt) [reply packetbeat/flows/worker_test.go:119]: It was failing the lint check saying this is not used.

File: packetbeat/flows/worker_test.go [hunk 3]
```diff
@@ -126,4 +126,27 @@ func TestCreateEvent(t *testing.T) {
...
+	// when enableDeltaFlowReporting is true, the flow stats should be reset
+	expectbiFlow := &biFlow{
+		id:       id.rawFlowID,
+		killed:   1,
+		createTS: start,
+		ts:       end,
+		dir:      flowDirForward,
+	}
+	expectbiFlow.stats[0] = &flowStats{uintFlags: []uint8{1, 1}, uints: []uint64{0, 0}}
+	expectbiFlow.stats[1] = &flowStats{uintFlags: []uint8{1, 1}, uints: []uint64{0, 0}}
```
Comment (haesbaert) [inline packetbeat/flows/worker_test.go:139]: I'd feel a bit more comfortable asserting that `bif.stats[0].uints` is not equal `expectbiFlow.stats[0].uints` here, in case the test above changes, I'll leave it at your discretion.

**Review by haesbaert — APPROVED** [review]
The typo on `cummlative` as pointed out, other that LGTM!

File: packetbeat/flows/worker_test.go (hunk 3 above)
Comment (kvalliyurnatt) [reply packetbeat/flows/worker_test.go:140]: done

**Comment by kvalliyurnatt** [comment]
@haesbaert any idea why the filebeat tests are failing, they don't seem to be related to my changes .

**Comment by haesbaert** [comment]
> @haesbaert any idea why the filebeat tests are failing, they don't seem to be related to my changes .

No idea, I'll ping some people, I don't think it should block this PR though.
